rest/handler.go

```diff
@@ -35,62 +38,65 @@ func CreateHandler(ctx context.Context, prometheusURL string) (http.Handler, err
 		return nil, err
 	}
 
+	ls := &langserverHandler{langserver: langserver}
+	ls.m = make(map[string]http.Handler)
+	ls.createDefaultHandlers()
+
 	return &langserverHandler{langserver: langserver}, nil
 }
 
+func (h *langserverHandler) createDefaultHandlers() {
+	httpRequestsTotal := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "http_requests_total",
+		Help: "Count of all HTTP requests",
+	}, []string{"code", "method"})
+
+	httpRequestDuration := prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name: "http_request_duration_seconds",
+		Help: "Duration of all HTTP requests base on endpoint being queried",
+	}, []string{"code", "handler"})
+
+	r := prometheus.NewRegistry()
```

Comment:
It'd be better if the registry would be passed as an optional argument to `CreateHandler`, since the handler might be used in a context where instrumentation is already in place by existing middleware (e.g. in https://github.com/prometheus/prometheus/pull/6872).

I guess it would be best to have both `func CreateHandler(ctx context.Context, prometheusURL string) (http.Handler, error)` and  `func CreateInstrumentedHandler(ctx context.Context, prometheusURL string, r *prometheus.Registry) (http.Handler, error)`.

Sorry for the back and forth.

Reply:
alright . it's fine.